packages/client/src/shared/lib/hooks/useAuth.ts

```diff
@@ -0,0 +1,14 @@
+import { getCurrentUser } from '@@entities/user';
+import { makeMapState, useMapState } from '../model/hooks';
+
+export const useAuth = () => {
+  const mapState = makeMapState((state) => ({
+    user: getCurrentUser(state),
+  }));
+
+  const { user } = useMapState(mapState);
+
+  const { isLoading, isAuth, data } = user;
```

Comment:
Можно декомпозировать чуть глубже
```suggestion
    const {
    user: { isLoading, isAuth, data },
  } = useMapState(mapState);
```